commit db3e664b1501593d21360526c034895e706cfaba
Author: Jean-Francois Dockes <jf@dockes.org>
Date:   Mon Oct 30 19:05:15 2023 +0100

    restore minimal python2 compat for the recoll extension python code

diff --git a/python/recoll/recoll/conftree.py b/python/recoll/recoll/conftree.py
index b32cbb2e..949c9f93 100644
--- a/python/recoll/recoll/conftree.py
+++ b/python/recoll/recoll/conftree.py
@@ -329,7 +329,10 @@ def valToBool(s):
 # Copyright 2019 Kenneth Reitz
 # Apache License - Version 2.0, January 2004 - http://www.apache.org/licenses
 
-from collections.abc import MutableMapping
+try:
+    from collections.abc import MutableMapping
+except:
+    from collections import MutableMapping
 
 class CaseInsensitiveDict(MutableMapping):
     """
diff --git a/src/python/recoll/recoll/rclconfig.py b/src/python/recoll/recoll/rclconfig.py
index bf4a435c..29ec9b8e 100644
--- a/python/recoll/recoll/rclconfig.py
+++ b/python/recoll/recoll/rclconfig.py
@@ -1,4 +1,5 @@
 #!/usr/bin/env python3
+from __future__ import print_function
 
 import locale
 import re
@@ -13,7 +14,7 @@ except:
     import conftree
 
 def msg(s):
-    print(f"{s}", file=sys.stderr)
+    print("%s" % s, file=sys.stderr)
     
 class RclDynConf:
     def __init__(self, fname):
@@ -101,7 +102,7 @@ class RclConfig:
             pass
         if f is None:
             raise(Exception(
-                f"Can't open default/system recoll.conf in [{baseconfdir}]. " +
+                "Can't open default/system recoll.conf in [%s]. " % baseconfdir +
                 "Please set RECOLL_DATADIR in the environment to point " +
                 "to the installed recoll data files."))
         else:
@@ -173,6 +174,6 @@ class RclExtraDbs:
     
 if __name__ == '__main__':
     config = RclConfig()
-    print(f"topdirs = {config.getConfParam('topdirs')}")
+    print("topdirs = %s" % config.getConfParam('topdirs'))
     extradbs = RclExtraDbs(config)
-    print(f"{extradbs.getActDbs()}")
+    print("%s" % extradbs.getActDbs())
